From 19814b7ad4ea6f0cc4cab34e50ebab2e180fc269 Mon Sep 17 00:00:00 2001
From: Harish Mahendrakar <harish.mahendrakar@ittiam.com>
Date: Mon, 16 Jan 2017 11:26:26 +0530
Subject: Decoder: Fixed DoS in header decode when no PPS is present

When the input does not contain PPS and decoder is in header decode
mode, decoder was entering an infinite loop.

Bug: 33621215
(cherry picked from commit 33e1b190d6db09bd72a9f0f51acef4b14eabd6ff)
(cherry picked from commit 2cd2f7a33535f29c87a1952e0d4b43a80b6f1f3c)
---
 decoder/ih264d_api.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/decoder/ih264d_api.c b/decoder/ih264d_api.c
index 4a81465..f1a0ea0 100644
--- a/decoder/ih264d_api.c
+++ b/decoder/ih264d_api.c
@@ -2083,6 +2083,7 @@ WORD32 ih264d_video_decode(iv_obj_t *dec_hdl, void *pv_api_ip, void *pv_api_op)
             {
                 /* a start code has already been found earlier in the same process call*/
                 frame_data_left = 0;
+                header_data_left = 0;
                 continue;
             }
 
-- 
cgit v1.1

